oak-run-commons/src/main/java/org/apache/jackrabbit/oak/index/indexer/document/flatfile/FlatFileStore.java

```diff
@@ -54,8 +57,18 @@ public FlatFileStore(BlobStore blobStore, File storeFile, NodeStateEntryReader e
         this.entryReader = entryReader;
         this.preferredPathElements = preferredPathElements;
         this.algorithm = algorithm;
+        this.metadataFile = metadataFile;
+    }
+
+    public FlatFileStore(BlobStore blobStore, File storeFile, NodeStateEntryReader entryReader, Set<String> preferredPathElements, Compression algorithm) {
+        this(blobStore, storeFile, null, entryReader, preferredPathElements, algorithm);
+    }
+
+    public String getStorePath() {
+        return storeFile.getParentFile().getAbsolutePath();
     }
 
+    @Deprecated
```

Comment:
for documentation purposes of deprecation - 

`
It is strongly recommended that the reason for deprecating a program element be explained in the documentation, using the @deprecated javadoc tag. The documentation should also suggest and link to a recommended replacement API, if applicable. A replacement API often has subtly different semantics, so such issues should be discussed as well.
It is recommended that a since value be provided with all newly annotated program elements. Note that since cannot be mandatory, as there are many existing annotations that lack this element value.
There is no defined order among annotation elements. As a matter of style, the since element should be placed first.
The @Deprecated annotation should always be present if the @deprecated javadoc tag is present, and vice-versa.`

Reply:
done